addons/road-generator/road_point.gd

```diff
@@ -458,37 +461,49 @@ func _exit_tree():
 			singling_rp_ref = null
 
 
-## Evaluates a RoadPoint's prior/next_pt_inits and verifies that they describe a
-## valid junction. A junction is valid if this RoadPoint agrees with what the
-## associated RoadPoint is saying. Invalid junctions are cleared. But, only if
-## auto_refresh is true.
+## Evaluates THIS RoadPoint's prior/next_pt_inits and verifies that they
+## describe a valid junction. A junction is valid if THIS RoadPoint agrees with
+## what the associated RoadPoint is saying. Invalid junctions are cleared. But,
+## only if auto_refresh is true.
 func validate_junctions(auto_refresh: bool):
 	if not auto_refresh:
 		return
 
 	var prior_point: RoadPoint
 	var next_point: RoadPoint
-	var eval_point: RoadPoint
-	var eval_pt_init: NodePath
 
-	# Get valid Prior and Next RoadPoints
+	# Get valid Prior and Next RoadPoints for THIS RoadPoint
 	if prior_pt_init and not prior_pt_init == "":
 		prior_point = get_node(prior_pt_init)
 	if next_pt_init and not next_pt_init == "":
 		next_point = get_node(next_pt_init)
 
-	# Verify Prior RoadPoint identifies this one as Next
+	# Clear invalid junctions
 	if is_instance_valid(prior_point):
-		eval_pt_init = prior_point.next_pt_init
-		if eval_pt_init and not eval_pt_init == "":
-			eval_point = get_node(eval_pt_init)
-			if not eval_point == self:
-				prior_pt_init = null
+		if not _is_junction_valid(prior_point):
+			prior_pt_init = null
+	if is_instance_valid(next_point):
+		if not _is_junction_valid(next_point):
+			next_pt_init = null
+
 
-	# Verify Next RoadPoint identifies this one as Prior
+## Evaluates INPUT RoadPoint's prior/next_pt_inits. Returns true if at least
+## one of them references THIS RoadPoint. Otherwise, returns false.
+func _is_junction_valid(point: RoadPoint)->bool:
```

Comment:
Something to discuss: how to make this scenario function ok. Basically, the narrow scenario where one roadpoint _doesn't_ have representation in what it's connected to, but what it's connected to does have "an open slot". basically the scenario here in this video where I'm trying to connect two roads together (which also happen to be facing the opposite way).

I even wonder if it's possible to create a unit test around this to ensure the expected behavior is there.

https://github.com/TheDuckCow/godot-road-generator/assets/2958461/497a67fa-10bf-4dee-bdb8-280e0eb9b311

Reply:
Two issues came up in our meeting discussion:
1. Can't manually connect two RoadPoints due to validation.
2. RoadSegment drawing is distorted when RoadPoints are rotated 180° from each other.

**Issue 1: Can't manually connect two RoadPoints due to validation**
It seemed like the conclusion was that this was a limitation of the requested features, which makes sense. The workaround was to disable auto_refresh. That was part of the requested features and it works.

We also discussed potentially adding buttons on a panel to more intuitively facilitate manually connecting RoadPoints without the need to disable auto_refresh.

Additionally, we could implement a rule that says, if the destination RoadPoint has an empty slot, then update both the source and destination RoadPoints. That would require a series of rules. Particularly, if either slot (Prev/Next) on the destination could be used.

**Issue 2: RoadSegment drawing is distorted when RoadPoints are rotated 180°**
Lane drawing logic is dependent on the orientation between the start and end points. Two RoadPoints must be properly rotated in order to comprise a valid RoadSegment for drawing. If two RoadPoints are rotated to face in opposite directions, then they will appear skewed. That is expected behavior.

We could overhaul the logic to make Segments agnostic to RoadPoint rotation. That would require some additional rules and effort. But, it could be done.

Under normal circumstances a Next connects to a Prior. In that scenario, the lane array order and physical rotations are aligned between the two RoadPoints. But, if we wanted to go crazy, we could say that Prior>Prior and Next>Next are also okay.

In those scenarios, we would have to reverse the order of one of the arrays to account for the fact that we are effectively connecting RoadPoints that are facing the opposite direction. Additionally, the drawing logic would have to take into account the roughly 180° rotational offset.